plugins/infinispan100/src/main/java/org/radargun/service/InfinispanRestAPI.java

```diff
@@ -23,46 +22,74 @@
  */
 public class InfinispanRestAPI {
 
+   protected final Log log = LogFactory.getLog(getClass());
+
    private ObjectMapper mapper;
-   private HttpClient httpClient;
-   private HttpResponse<String> response;
    private Integer defaultPort;
    private static final String CACHE_MANAGER_RESOURCE = "v2/cache-managers/clustered";
    private String serverIp;
-   private final Log log = LogFactory.getLog(getClass());
 
    public InfinispanRestAPI(Integer defaultPort) {
-      mapper = new ObjectMapper();
-      mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-      mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
-      httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1).build();
+      this.mapper = new ObjectMapper();
+      this.mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+      this.mapper.setPropertyNamingStrategy(PropertyNamingStrategy.SNAKE_CASE);
       this.defaultPort = defaultPort;
-      try {
-         serverIp = Inet4Address.getLocalHost().getHostAddress();
-      } catch (UnknownHostException e) {
-         log.error("Wasn't possible to get HostAddress to execute Rest Operations", e);
-         throw new RuntimeException(e);
-      }
+      this.serverIp = lookupServerIp();
    }
 
-   public CacheManagerInfo getCacheManager() {
-      String url = String.format("http://%s:%s/rest/%s", serverIp, this.defaultPort, CACHE_MANAGER_RESOURCE);
-      HttpRequest get = HttpRequest.newBuilder()
-            .uri(URI.create(url))
-            .GET()
-            .build();
+   private String lookupServerIp() {
+      // the following are predefined values in the INFINISPAN10 XML Files
+      String interfaceName = System.getProperty("radargun.infinispan.interfaceName", "public");
+      String bindAddress = System.getProperty("infinispan.bind.address", "127.0.0.1");
+
+      // given a config, search the network address
+      NetworkAddress networkServerAddress = NetworkAddress.fromString(interfaceName, bindAddress);
+      String host = networkServerAddress.getAddress().getHostAddress();
+
+      log.info(String.format("interface: %s, bindAddress: %s, host: %s", interfaceName, bindAddress, host));
+
+      return host;
+   }
 
+   public CacheManagerInfo getCacheManager() {
+      // network is not reliable
+      int count = 0;
       CacheManagerInfo cacheManagerInfo = null;
-      try {
-         response = httpClient.send(get, HttpResponse.BodyHandlers.ofString());
-         String json = response.body();
-         cacheManagerInfo = mapper.readValue(json, CacheManagerInfo.class);
-      } catch (IOException e) {
-         e.printStackTrace();
-      } catch (InterruptedException e) {
-         e.printStackTrace();
+      while (cacheManagerInfo == null && count++ < 5) {
```

Comment:
Would it make more sense to add this retry logic to the `doGet` method directly? It seems like this will be useful for any REST API calls.

Reply:
Done